### ISSUE TITLE
Edit collection page is not loading anymore after page refresh

### ISSUE THREAD

**Issue by AlexandraMoga** [mozilla/addons-frontend#5022]
STR:

1. Log in to AMO dev
2. Go to My Collections from the user menu and select one of your collections to view its details page
3. Click on Edit Collection
4. Refresh the page, then click on Save Collection

Actual result:
The list of add-ons and the collection details are not loading 

Expected result:
The list of add-ons and the collection details are loaded 

Notes:
- another valid way to reproduce this is by manually changing the page in the URL after clicking on Edit Collection
- also reproduced when Edit Collection link is opened in a new tab
- reproduced on AMO-dev with FF60, Win 10x64

![coll page load](https://user-images.githubusercontent.com/31961530/40118242-0ac851c6-5922-11e8-9f15-47669d8c6be7.gif)

**Comment by bobsilverberg** [comment]
I finally figured out the source of this problem: We had some code in place that would fetch the collection add-ons for a collection when a user switched into edit mode. I introduced this to address the case when a user is viewing page 2 of a collection, and switches into edit mode. I assumed the best thing to do would be to start them off at page 1, so I re-fetched the add-ons (where page 1 would be the default).

This was causing a problem with SSR because, I believe, the fetch was being dispatched on the server, but the load wasn't happening in time so the saga was not completing successfully. After many different tries at fixing this I realized a better solution would be to not re-fetch the add-ons at all, and just bring the user to the edit screen with the current page's add-ons still loaded. This has an added performance benefit.

I'll be opening a PR with the code that does this.
